tests/test_grid.py

```diff
@@ -0,0 +1,169 @@
+"""Test chi^2 gridding routines"""
+import io
+import os
+import tempfile
+
+import astropy.units as u
+import numpy as np
+
+import pint.config
+import pint.gridutils
+import pint.models.parameter as param
+import pint.residuals
+from pint.fitter import GLSFitter
+from pint.models.model_builder import get_model, get_model_and_toas
+from pint.toa import get_TOAs
+
+
+def test_grid_singleprocessor():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = GLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 3 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 3 * f.model.F0.uncertainty,
+        15,
+    )
+    F1 = np.linspace(
+        f.model.F1.quantity - 3 * f.model.F1.uncertainty,
+        f.model.F1.quantity + 3 * f.model.F1.uncertainty,
+        17,
+    )
+
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1"), (F0, F1), ncpu=1)
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_multiprocessor():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = GLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 3 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 3 * f.model.F0.uncertainty,
+        15,
+    )
+    F1 = np.linspace(
+        f.model.F1.quantity - 3 * f.model.F1.uncertainty,
+        f.model.F1.quantity + 3 * f.model.F1.uncertainty,
+        17,
+    )
+
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1"), (F0, F1))
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_oneparam():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = GLSFitter(t, m)
```

Comment:
Why use the GLSFitter? Isn't it going to be slower and more numerically unstable? And not needed here?